activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java

```diff
@@ -593,14 +593,71 @@ public void shouldGetProcessInstancesFilteredByAppVersion() {
                 .aRunningProcessInstanceWithAppVersion("second", "2");
         processInstanceBuilder.aRunningProcessInstanceWithAppVersion("third", "3");
         eventsAggregator.sendAll();
-        
+
         List<String> processInstanceIds = List.of(processInstanceAppVersion1.getId(),
                 processInstanceAppVersion2.getId());
 
         shouldGetProcessInstancesFilteredBySingleValue(processInstanceAppVersion1.getId(), "appVersion=1" );
         shouldGetProcessInstancesFilteredByList(processInstanceIds, "appVersion=1,2");
     }
 
+    @Test
+    public void shouldGetProcessInstancesFilteredBySuspendedDate() {
+        //given
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+        sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+        Date suspendedDateToday = new Date();
+        Date suspendedDateTwoDaysAgo = new Date();
+        Date suspendedDateFiveDaysAfter = new Date();
+
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        Date now = cal.getTime();
+
+        //Start a process and set it's suspended date as current date
+        suspendedDateToday.setTime(now.getTime());
+        ProcessInstance processInstanceSuspendedToday = processInstanceBuilder
+            .aRunningProcessInstanceWithSuspendedDate("suspendedDateToday", suspendedDateToday);
+
+        //Start a process and set it's suspended date as current date minus two days
+        suspendedDateTwoDaysAgo.setTime(now.getTime() - Duration.ofDays(2).toMillis());
+        processInstanceBuilder.aRunningProcessInstanceWithSuspendedDate("suspendedDateTwoDaysAgo", suspendedDateTwoDaysAgo);
+
+        //Start a process and set it's suspended date as current date plus five days
+        suspendedDateFiveDaysAfter.setTime(now.getTime() + Duration.ofDays(5).toMillis());
+        processInstanceBuilder.aRunningProcessInstanceWithSuspendedDate("suspendedDateFiveDaysAfter", suspendedDateFiveDaysAfter);
+
+        eventsAggregator.sendAll();
+
+        await().untilAsserted(() -> {
+
+            //when
+            //set from date to yesterday date
+            Date fromDate = new Date(now.getTime() - Duration.ofDays(1).toMillis());
+            // to date, from date plus 2 days
+            Date toDate = new Date(now.getTime() + Duration.ofDays(2).toMillis());
+            //when
+            ResponseEntity<PagedModel<ProcessInstanceEntity>> responseEntityFiltered = testRestTemplate
```

Comment:
In addition to the range, could you also cover the case of using a specific suspended date (`?suspendedDate=`)?

Reply:
Thanks, this turned out a bug into `startDate`, `completedDate` and so the `suspendedDate` due to a missing annotation with variable declaration in `ProcessInstanceEntity.java` which is fixed and more test cases added to support static date query filtering.